lang_qc/db/helper/qc.py

```diff
@@ -0,0 +1,92 @@
+# Copyright (c) 2023 Genome Research Ltd.
+#
+# Authors:
+#  Kieron Taylor <kt19@sanger.ac.uk>
+#
+# This file is part of npg_langqc.
+#
+# npg_langqc is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the Free Software
+# Foundation; either version 3 of the License, or (at your option) any later
+# version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program. If not, see <http://www.gnu.org/licenses/>.
+
+from typing import Dict, List
+
+from pydantic import BaseModel, Field
+from sqlalchemy import select
+from sqlalchemy.orm import Session, selectinload
+
+from lang_qc.db.qc_schema import QcState as QcStateDb
+from lang_qc.db.qc_schema import QcStateDict, QcType, SeqProduct, User
+from lang_qc.models.qc_state import QcState
+
+
+class BulkQcFetch(BaseModel):
+
+    session: Session = Field(
+        title="SQLAlchemy Session",
+        description="A SQLAlchemy Session for the LangQC database",
+    )
+
+    class Config:
+        arbitrary_types_allowed = True
+
+    def query_by_id_list(self, ids: List) -> Dict[str, Dict[str, QcState]]:
+        """
+        Fetches sequencing products with matching product IDs (checksums)
+
+        ids - a list of SHA256 strings generated by npg_id_generation
+
+        Returns a list of SeqProduct results with pre-loaded joins for QC
+        state reporting.
+        """
```

Comment:
Could we explain here that not finding a matching QC outcome is not an error and that if no matching data is found, an empty dict is returned.